### PR TITLE
Implement in-place inject-as-head

### PR DESCRIPTION
**Pull request by NicolasT**

### CODE REVIEW

**Comment by NicolasT** [comment]
```
test_inject_as_head : asserts shortcut for those who don't want to collapse periodically. (eta: 20 s) ... ok
arakoon_system_tests.server.shaky.test_inject_as_head.test_inject_as_head_current_corrupt ... ok
Test inject_as_head with a fake database file (should fail) ... ok
test_inject_as_head : asserts shortcut for those who don't want to collapse periodically. (eta: 20 s) ... ok
arakoon_system_tests.server.shaky.test_inject_as_head.test_inplace_inject_as_head ... ok
```

File: src/node/dump_store.ml [hunk 1]
```diff
@@ -48,7 +48,7 @@ let dump_store filename =
 
 exception ExitWithCode of int;;
 
-let inject_as_head fn node_id cfg_fn force =
+let inject_as_head fn node_id cfg_fn force ~in_place =
```
Comment (domsj) [inline src/node/dump_store.ml]: Make force a labeled argument too?
The way you did it with ~in_place does add clarity at the call site.

File: src/node/dump_store.ml [hunk 2]
```diff
@@ -108,8 +108,18 @@ let inject_as_head fn node_id cfg_fn force =
       | None -> failwith "can't happen"
       | Some i -> Sn.to_int (Tlc2.get_file_number i)
     in
-    Lwt_io.printf "cp %S %S" fn old_head_name >>=fun () ->
-    File_system.copy_file fn old_head_name true >>= fun () ->
+    begin
+      if (not in_place)
+      then begin
+        Lwt_io.printf "cp %S %S" fn old_head_name >>=fun () ->
+        File_system.copy_file fn old_head_name true
```
Comment (domsj) [inline src/node/dump_store.ml]: I know this line is the same as before, but maybe we should make the last argument here a labeled one too?

**Comment by domsj** [comment]
Looks good

**Comment by NicolasT** [comment]
Are you OK if I add the labels & the thing keeps compiling?

**Comment by domsj** [comment]
Yes

**Comment by NicolasT** [comment]
Superseded by #315.

**Pull request closed without merging**